src/java.base/share/classes/java/lang/ThreadGroup.java

```diff
@@ -658,10 +658,17 @@ private List<ThreadGroup> synchronizedSubgroups() {
 
     /**
      * Returns a snapshot of the subgroups as an array, used by JVMTI.
+     * WARNING: Make sure this method does not trigger any class loading,
+     * because a ClassPrepareEvent can deadlock the debugger and debug agent.
```

Comment:
The comment is okay but if you are doing any further edits then you could change this to say that a callback for the ClassPrepare event can deadlock (the event is "ClassPrepare" in the JVM TI spec, I realize the "ClassPrepareEvent" may have come from JDI).